### PR TITLE
Fixes #26325 - Remove data-id aid attributes

### PR DESCRIPTION
**Pull request by tbrisker**
These were used by QE automation to identify links but are no longer
needed since https://github.com/SatelliteQE/airgun/commit/8438e7274a73e01dcb7073f47068eba877e2bed5

### CODE REVIEW

File: test/controllers/architectures_controller_test.rb [hunk 1]
```diff
@@ -11,16 +11,6 @@ class ArchitecturesControllerTest < ActionController::TestCase
   basic_pagination_per_page_test
   basic_pagination_rendered_test
 
-  def test_new_submit_button_id
```
Comment (tbrisker) [inline test/controllers/architectures_controller_test.rb:14]: These three tests aren't useful at all, the proper rendering is tested by `basic_new_test` and `basic_edit_test`

File: webpack/assets/javascripts/react_app/components/Layout/components/TaxonomyDropdown.js [hunk 2]
```diff
@@ -47,7 +47,7 @@ const TaxonomyDropdown = ({
         <li key={i}>
           <a
             className={`${taxonomyType.toLowerCase()}_menuitem`}
-            id={`aid_taxonomy_${taxonomy.title}`}
+            id={`select_taxonomy_${taxonomy.title}`}
```
Comment (tbrisker) [inline webpack/assets/javascripts/react_app/components/Layout/components/TaxonomyDropdown.js:50]: changed the id for the taxonomy selector to make sure we can clean up all that aid references everywhere.

**Comment by tbrisker** [comment]
test failure unrelated

**Comment by tbrisker** [comment]
found it - https://github.com/theforeman/foreman-infra/pull/999

**Review by timogoebel — APPROVED** [review]

**Comment by timogoebel** [comment]
Thanks, @tbrisker.

**Pull request merged**